### DISCUSSION TITLE
errors on android 13

### DISCUSSION THREAD

**Discussion by Chuanhuan** [RuriOSS/daijin#29] [General]
when I execute `container -run`. I got an error message.  I also tried to reinstall `tsu` and it's not working. 

```
proot error: execve("/usr/bin/su"): Function not implemented
proot info: possible causes:
  * the program is a script but its interpreter (eg. /bin/sh) was not found;
  * the program is an ELF but its interpreter (eg. ld-linux.so) was not found;
  * the program is a foreign binary but qemu was not specified;
  * qemu does not work correctly (if specified);
  * the loader was not found or doesn't work.
fatal error: see `proot --help`.
proot error: can't chmod '/data/data/com.termux/files/usr/tmp/proot-6270-eqRBQz': Function not implemented
proot error: can't chdir to '/root': Function not implemented

```

**Comment by Moe-hacker** [comment]
> when I execute `container -run`. I got an error message. I also tried to reinstall `tsu` and it's not working.
> 
> ```
> proot error: execve("/usr/bin/su"): Function not implemented
> proot info: possible causes:
>   * the program is a script but its interpreter (eg. /bin/sh) was not found;
>   * the program is an ELF but its interpreter (eg. ld-linux.so) was not found;
>   * the program is a foreign binary but qemu was not specified;
>   * qemu does not work correctly (if specified);
>   * the loader was not found or doesn't work.
> fatal error: see `proot --help`.
> proot error: can't chmod '/data/data/com.termux/files/usr/tmp/proot-6270-eqRBQz': Function not implemented
> proot error: can't chdir to '/root': Function not implemented
> ```

Sorry that can you tell me more info about this error？Seems that /usr/bin/su does not exist, but I used /bin/su in termux-container, that's strange. And if your device is rooted, it's recommended to use chroot container.

**Reply by Chuanhuan** [reply]
thx for replying. 

I have this error while installing Debian.

```
Downloaded 81.7939 Megabyte(s) in 7 second(s). (10931.52 KB/s)
[] EXTRACT ROOTFS...
tar: ./usr/bin/perlthanks: Cannot hard link to ‘./usr/bin/perlbug’: Permission denied
tar: ./usr/bin/perl5.36.0: Cannot hard link to ‘./usr/bin/perl’: Permission denied
tar: ./usr/bin/uncompress: Cannot hard link to ‘./usr/bin/gunzip’: Permission denied
81.8MiB 0:00:10 [7.74MiB/s] [===========>] 100%
tar: Exiting with failure status due to previous errors
[] REMOVE DOWNLOADED ROOTFS...
[] WRITE CONFIG FILE...
[] FIXING...
[] FIX DNS...
[] ADD SYSTEM GROUPS...
[] DONE!

```

I use proot and also see the reddit and someone also posted [chroot permission issue](https://www.reddit.com/r/termux/comments/117bn3j/termux_container_error/). 

Not sure is android 13 file Permission policy cause this issue?

**Reply by whyakari** [reply]
here everything works normally. i am using android pixelos 13.

**Comment by Xynonners** [comment]
same issue here, but /bin/su instead

EDIT: seems to be limited to certain rootfs
